core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -50,7 +50,13 @@ class KryoSerializer(conf: SparkConf)
   with Logging
   with Serializable {
 
-  private val bufferSizeKb = conf.getSizeAsKb("spark.kryoserializer.buffer", "64k")
+  private val bufferSize = conf.get("spark.kryoserializer.buffer", "64k")
+  private val bufferSizeKb = _
+  if (bufferSize.endsWith("k") || bufferSize.endsWith("kb")) {
+    bufferSizeKb = conf.getSizeAsKb("spark.kryoserializer.buffer", "64k")
+  } else {
+    bufferSizeKb = conf.getSizeAsMb("spark.kryoserializer.buffer", "64k")
```

Comment:
Let me wait for Debasish to confirm whether he can reproduce the error before updating this PR.